apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.UnitTests/PriorityListTests.cs

```diff
@@ -0,0 +1,235 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Google.Cloud.Spanner.V1;
+using Xunit;
+
+namespace Google.Cloud.Spanner.Data.Tests
+{
+    public class PriorityListTests
+    {
+        [Fact]
+        public void CanAddMultipleOfSamePriority()
+        {
+            PriorityList<TestItem> priorityList = new PriorityList<TestItem>();
+            TestItem item1, item2;
+            priorityList.Add(item1 = new TestItem(5));
+            priorityList.Add(item2 = new TestItem(5));
+            Assert.Equal(2, priorityList.Count);
+            AssertExist(priorityList, item1, item2);
+        }
+
+        [Fact]
+        public void DeleteRemovesCorrectItem()
+        {
+            PriorityList<TestItem> priorityList = new PriorityList<TestItem>();
+            TestItem toRemove;
+            priorityList.Add(toRemove = new TestItem(5));
+            priorityList.Add(new TestItem(5));
+            priorityList.Remove(toRemove);
+            Assert.Equal(1, priorityList.Count);
+            AssertNotExist(priorityList, toRemove);
+        }
+
+        [Fact]
+        public void DeleteRemovesOutOfSyncItem()
+        {
+            PriorityList<TestItem> priorityList = new PriorityList<TestItem>();
+            TestItem toRemove;
+            priorityList.Add(toRemove = new TestItem(5));
+            priorityList.Add(new TestItem(5));
+            toRemove.Priority = 4;
+
+            priorityList.Remove(toRemove);
+            Assert.Equal(1, priorityList.Count);
+            AssertNotExist(priorityList, toRemove);
+        }
+
+        [Fact]
+        public void DeleteNonExistentItemNoops()
+        {
+            PriorityList<TestItem> priorityList = new PriorityList<TestItem>();
+            priorityList.Add(new TestItem(1));
+            priorityList.Add(new TestItem(2));
+            priorityList.Remove(new TestItem(1));
+
+            Assert.Equal(2, priorityList.Count);
+        }
+
+        [Fact]
+        public void DeleteRemovesEventHook()
+        {
+            PriorityList<TestItem> priorityList = new PriorityList<TestItem>();
+            TestItem toRemove;
+            priorityList.Add(toRemove = new TestItem(5));
+            priorityList.Add(new TestItem(5));
```

Comment:
Adding this item doesn't seem necessary, unless you want to also test below that the item remaining in the list still has `HasEventHooks` true after the remove.